ludwig/schema/model_config.py

```diff
@@ -264,13 +257,30 @@ def _upgrade_config(config_dict: ModelConfigDict) -> ModelConfigDict:
         return upgrade_config_dict_to_latest_version(config_dict)
 
     @staticmethod
-    def _validate_config(config_dict: ModelConfigDict) -> None:
-        """Helper function used to validate the config using the Ludwig Schema.
+    def _validate_config(comprehensive_config: ModelConfigDict) -> None:
+        """Helper function used to validate the config using the Ludwig Schema and additional checks.
 
         Args:
             config_dict: Config Dictionary
         """
-        validate_config(config_dict)
+        # Schema validation.
+        check_schema(comprehensive_config)
+
+        # Additional checks.
+        auxiliary_checks.check_feature_names_unique(comprehensive_config)
```

Comment:
Nit: for brevity, you could create a registry of auxiliary checks

Reply:
Done.